src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/rollup/RollupRunner.kt

```diff
@@ -277,19 +279,37 @@ object RollupRunner : ScheduledJobRunner,
             while (rollupSearchService.shouldProcessRollup(updatableJob, metadata)) {
                 do {
                     try {
-                        val rollupResult = when (val rollupSearchResult = rollupSearchService.executeCompositeSearch(updatableJob, metadata)) {
-                            is RollupSearchResult.Success -> {
-                                val compositeRes: InternalComposite = rollupSearchResult.searchResponse.aggregations.get(updatableJob.id)
-                                metadata = metadata.incrementStats(rollupSearchResult.searchResponse, compositeRes)
-                                when (val rollupIndexResult = rollupIndexer.indexRollups(updatableJob, compositeRes)) {
-                                    is RollupIndexResult.Success -> RollupResult.Success(compositeRes, rollupIndexResult.stats)
-                                    is RollupIndexResult.Failure -> RollupResult.Failure(rollupIndexResult.message, rollupIndexResult.cause)
+                        val roles = if (job.user == null) {
+                            settings.getAsList("", listOf("all_access"))
+                        } else {
+                            job.user.roles
+                        }
```

Comment:
minor: move this to a method `Rollup.getRoles(): List<String> { .. }` and just use `job.geRoles()` here and other places fetching roles

Reply:
good idea, refactored